### PR TITLE
feat: add task retry scheduler

### PR DESCRIPTION
**Pull request by will-2012**
### Description

1. Add a task retry scheduler, ensuring objects can be sealed or reject-unsealed(the memory task will be aborted when it has been retried several times, and the task retry scheduler can retry this aborted kind of task).
2. Object can be accessed when the payload is uploaded in the primary sp, and no need to wait for sealed.
3. Anyone can use the query upload progress interface.

### Rationale

The task retry scheduler scans the recent upload context and re-push upload context to the dispatcher. If a task has not been sealed for a long time, it will be reject-unsealed.

### Example

```toml
[Manager]
# optional, default is false
EnableTaskRetryScheduler = true

# optional, default is 86400
# RejectUnsealThresholdSecond is as the following meanings:
# retry replicate and seal task when the task's create_timestamp + RejectUnsealThresholdSecond > now.time();
# retry reject unseal when the task's create_timestamp + RejectUnsealThresholdSecond <= now.time() && 
# create_timestamp + 2*RejectUnsealThresholdSecond > now.time().
RejectUnsealThresholdSecond = 86400
```
The above options turn on the task retry scheduler.
### Changes

Notable changes: 
* Add task retry scheduler which includes background retry workflow.
* Update get object's auth workflow.
* Update query upload progress's auth workflow.

### Potential Impacts
* Database pressure.

### CODE REVIEW

File: modular/authenticator/authenticator_test.go [hunk 1]
```diff
@@ -4,6 +4,9 @@ import (
 	"context"
 	"encoding/hex"
 	"errors"
+	"strings"
+	"testing"
+	"time"
 
 	sdkmath "cosmossdk.io/math"
 	"github.com/bnb-chain/greenfield-storage-provider/base/gfspapp"
```
Comment (sysvm) [inline modular/authenticator/authenticator_test.go:12]: sort imported pkgs

File: modular/authenticator/authenticator_test.go [hunk 2]
```diff
@@ -21,10 +24,6 @@ import (
 
 	"google.golang.org/grpc"
```
Comment (sysvm) [inline modular/authenticator/authenticator_test.go:25]: ditto

File: modular/manager/manage_task.go [hunk 3]
```diff
@@ -10,6 +10,7 @@ import (
 	"time"
 
 	"cosmossdk.io/math"
+	virtualgrouptypes "github.com/bnb-chain/greenfield/x/virtualgroup/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"golang.org/x/exp/slices"
```
Comment (sysvm) [inline modular/manager/manage_task.go:15]: ditto

**Review by alexgao001 — APPROVED** [review]

File: modular/authenticator/authenticator_test.go (hunk 1 above)
Comment (will-2012) [reply modular/authenticator/authenticator_test.go:19]: Fixed.

File: modular/authenticator/authenticator_test.go (hunk 2 above)
Comment (will-2012) [reply modular/authenticator/authenticator_test.go]: Fixed.

File: modular/manager/manage_task.go (hunk 3 above)
Comment (will-2012) [reply modular/manager/manage_task.go]: Fixed.

**Review by sysvm — APPROVED** [review]
LGTM

**Review by ruojunm — APPROVED** [review]

**Pull request merged**